src/main/java/org/betonquest/betonquest/commands/QuestCommand.java

```diff
@@ -1813,6 +1829,135 @@ private Optional<List<String>> completeDownload(final String... args) {
+                    .forEach(entry -> sender.sendMessage("§b- " + entry.getKey() + "§e: §a" + entry.getValue()));
+            return;
+        }
+        switch (args[3].toLowerCase(Locale.ROOT)) {
+            case "set":
+            case "s":
+                if (args.length == 4) {
+                    log.debug("Missing amount");
+                    sendMessage(sender, "arguments");
+                    return;
+                }
+                final String value;
+                if (args.length == 5) {
+                    value = null;
+                } else {
+                    value = String.join(" ", Arrays.copyOfRange(args, 5, args.length));
+                }
+                log.debug("Setting value " + value + " for key " + args[4] + " for " + profile + " in " + variableObjective.getLabel());
+                variableObjective.store(profile, args[4], value);
+                sendMessage(sender, "value_set", value, args[4]);
+                break;
+            default:
+                // if there was something else, display error message
+                log.debug("The argument was unknown");
+                sendMessage(sender, "unknown_argument");
+                break;
+        }
+    }
+
+    /**
+     * Returns a list including all possible options for tab complete of the
+     * /betonquest variables command
+     *
+     * @param args
+     * @return
+     */
+    private Optional<List<String>> completeVariables(final String... args) {
+        if (args.length == 2) {
+            return Optional.empty();
+        }
+        if (args.length == 3) {
+            final String last = args[args.length - 1];
+            if (last == null || !last.contains(".")) {
+                return completePackage();
+            } else {
+                final String pack = last.substring(0, last.indexOf('.'));
+                final QuestPackage configPack = Config.getPackages().get(pack);
+                if (configPack == null) {
+                    return Optional.of(new ArrayList<>(0));
+                }
+                final ConfigurationSection configuration = configPack.getConfig().getConfigurationSection("objectives");
+                final List<String> completions = new ArrayList<>();
+                if (configuration != null) {
+                    for (final String key : configuration.getKeys(false)) {
+                        final String rawObjectiveInstruction = configuration.getString(key);
+                        if (rawObjectiveInstruction != null && rawObjectiveInstruction.stripIndent().startsWith("variable")) {
+                            completions.add(pack + '.' + key);
+                        }
+                    }
+                }
+                return Optional.of(completions);
+            }
+        }
+        if (args.length == 4) {
+            return Optional.of(Arrays.asList("list", "set"));
+        }
+        return Optional.of(new ArrayList<>());
```

Comment:
Use either `List.of()` or even better `Collections.emptyList()` for better performance.